projects/plugins/jetpack/extensions/blocks/premium-content/edit.js

```diff
@@ -82,148 +57,21 @@ const defaultString = null;
-		const minPrice = minimumTransactionAmountForCurrency( attributes.newPlanCurrency );
-		const minimumPriceNote = sprintf(
-			// translators: %s: Price
-			__( 'Minimum allowed price is %s.', 'jetpack' ),
-			formatCurrency( minPrice, attributes.newPlanCurrency )
-		);
-
-		if ( newPrice < minPrice ) {
-			onError( props, minimumPriceNote );
-			callback( false );
-			return;
-		}
-
-		if ( ! isPriceValid( attributes.newPlanCurrency, newPrice ) ) {
-			onError( props, __( 'Plan requires a valid price', 'jetpack' ) );
-			callback( false );
-			return;
-		}
-
-		const data = {
-			currency: attributes.newPlanCurrency,
-			price: attributes.newPlanPrice,
-			title: attributes.newPlanName,
-			interval: attributes.newPlanInterval,
-		};
-		const fetch = { path, method, data };
-		apiFetch( fetch ).then(
-			/**
-			 * @param { any } result - Result of fetch query
-			 * @returns { void }
-			 */
-			result => {
-				/**
-				 * @type { Plan }
-				 */
-				const newProduct = {
-					id: result.id,
-					title: result.title,
-					interval: result.interval,
-					price: result.price,
-					currency: result.currency,
-				};
-				setProducts( products.concat( [ newProduct ] ) );
-				// After successful adding of product, we want to select it. Presumably that is the product user wants.
-				selectPlan( newProduct );
-				onSuccess( props, __( 'Successfully created plan', 'jetpack' ) );
-				if ( callback ) {
-					callback( true );
-				}
-			},
-			/**
-			 * @returns { void }
-			 */
-			() => {
-				onError( props, __( 'There was an error when adding the plan.', 'jetpack' ) );
-				if ( callback ) {
-					callback( false );
-				}
-			}
-		);
-	}
-
-	/**
-	 * @param {Plan} plan - plan whose description will be retrieved
-	 * @returns {?string} Plan description with price.
-	 */
-	function getPlanDescription( plan ) {
-		const amount = formatCurrency( parseFloat( plan.price ), plan.currency );
-		if ( plan.interval === '1 month' ) {
-			return sprintf(
-				// translators: %s: amount
-				__( '%s / month', 'jetpack' ),
-				amount
-			);
-		}
-		if ( plan.interval === '1 year' ) {
-			return sprintf(
-				// translators: %s: amount
-				__( '%s / year', 'jetpack' ),
-				amount
-			);
-		}
-		if ( plan.interval === 'one-time' ) {
-			return amount;
-		}
-		return sprintf(
-			// translators: %s: amount, plan interval
-			__( '%1$s / %2$s', 'jetpack' ),
-			amount,
-			plan.interval
-		);
-	}
-
-	/**
-	 * @param {Plan} plan - selected plan
-	 */
-	function selectPlan( plan ) {
-		props.setAttributes( { selectedPlanId: plan.id } );
-	}
-	//We would like to hide the tabs and controls when user clicks outside the premium content block
-	/**
-	 * @type { ContainerRef }
-	 */
-	const wrapperRef = useRef( null );
-	useOutsideAlerter( wrapperRef, hasSelectedInnerBlock );
-
-	const { isSelected, className } = props;
+	const setSelectedProductId = productId => setAttributes( { selectedPlanId: productId } );
```

Comment:
I'm not a fan of passing this function around, but I couldn't come up with a better way of sharing `setAttributes` (with a variable attribute name — all payment blocks use a different name...) with the product management component and its state. 🤔